.github/workflows/main.yml

```diff
@@ -28,23 +28,23 @@ jobs:
       - uses: "actions/setup-python@v2"
         with:
           python-version: "${{ matrix.python-version }}"
+
       - name: "Install dependencies"
         run: |
           set -xe
           python -VV
           python -m site
           python -m pip install --upgrade pip setuptools wheel
           python -m pip install --upgrade virtualenv tox tox-gh-actions
+          pip install pytest-cov
+          pip install pytest
+          pip install coveralls
+
       - name: "Run tox targets for ${{ matrix.python-version }}"
         run: "python -m tox"
 
-      # - name: "Report to coveralls"
-      #   # coverage is only created in the py39 environment
-      #   # --service=github is a workaround for bug
-      #   # https://github.com/coveralls-clients/coveralls-python/issues/251
-      #   if: "matrix.python-version == '3.9'"
-      #   run: |
-      #     pip install coveralls
-      #     coveralls --service=github
-      #   env:
-      #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: "Report to coveralls"
+        if: matrix.python-version == 3.9
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: coveralls debug --service=github
```

Comment:
@solancer Thanks for trying so patiently!

You activated the debug mode, which means that all output is printed to the console and no attempt is made to send it - this is unfortunately the reason why the build succeeds :-/

See here
https://coveralls-python.readthedocs.io/en/latest/troubleshooting.html